runtime/src/accounts_db.rs

```diff
@@ -10145,12 +10151,125 @@ pub mod tests {
         );
     }
 
+    #[test]
+    fn test_accountsdb_scan_snapshot_stores_hash_not_stored() {
+        solana_logger::setup();
+        let accounts_db = AccountsDb::new_single_for_tests();
+        let (storages, raw_expected) = sample_storages_and_accounts(&accounts_db);
+        storages.iter().for_each(|storage| {
+            accounts_db.storage.remove(&storage.slot(), false);
+        });
+
+        let hash = Hash::default();
+
+        // replace the sample storages, storing default hash values so that we rehash during scan
+        let storages = storages
+            .iter()
+            .map(|storage| {
+                let slot = storage.slot();
+                let copied_storage = accounts_db.create_and_insert_store(slot, 10000, "test");
+                let all_accounts = storage
+                    .all_accounts()
+                    .iter()
+                    .map(|acct| (*acct.pubkey(), acct.to_account_shared_data()))
+                    .collect::<Vec<_>>();
+                let accounts = all_accounts
+                    .iter()
+                    .map(|stored| (&stored.0, &stored.1))
+                    .collect::<Vec<_>>();
+                let slice = &accounts[..];
+                let account_data = (slot, slice, INCLUDE_SLOT_IN_HASH_TESTS);
```

Comment:
For the two tests (this one and the following one), should we also include the test case where INCLUDE_SLOT_IN_HASH_TESTS is a different value and expect a different outcome?

Reply:
done